standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/RetryingMetaStoreClient.java

```diff
@@ -264,17 +238,6 @@ public Object run() throws MetaException {
     return ret;
   }
 
-  private static boolean isRecoverableMetaException(MetaException e) {
-    String m = e.getMessage();
-    if (m == null) {
-      return false;
-    }
-    if (m.contains("java.sql.SQLIntegrityConstraintViolationException")) {
-      return false;
-    }
-    return IO_JDO_TRANSPORT_PROTOCOL_EXCEPTION_PATTERN.matcher(m).matches();
```

Comment:
TTransportException can be wrapped as MetaException, So I think we should keep this line.

Reply:
@saihemanth-cloudera nice concern! And thanks for your review.
IMO we should seperate TTransportException from MetaException, because they are responsible for different things:

- TTransportException means the exception in transport, like network issue.
- MetaException means the meta data exception ,like meta data corrupted.

BTW could you explain in which case TTransportException will be wrapped as MetaException?

Reply:
note, there is no constructor in MetaException to supply the root cause